mne/datasets/fetch.py

```diff
@@ -0,0 +1,244 @@
+# Authors: Adam Li <adam2392@gmail.com>
+#
+# License: BSD Style.
+
+import sys
+import os
+import os.path as op
+from distutils.version import LooseVersion
+from shutil import rmtree
+
+from .config import (_bst_license_text, RELEASES,
+                     TESTING_VERSIONED, MISC_VERSIONED, MNE_DATASETS)
+from .. import __version__ as mne_version
+from ..utils import (logger, warn, _safe_input, _soft_import)
+from .utils import _dataset_version, _do_path_update
+
+
+_FAKE_VERSION = None  # used for monkeypatching while testing versioning
+
+
+def fetch_dataset(dataset_params, processor=None, path=None,
+                  force_update=False, update_path=True, download=True,
+                  check_version=False, return_version=False, accept=False,
+                  auth=None):
+    """Fetch an MNE-compatible dataset.
+
+    Parameters
+    ----------
+    dataset_params : dict of dict
+        The dataset name and corresponding parameters to download the dataset.
+        The dataset parameters that contains the following keys:
+        ``archive_name``, ``url``, ``folder_name``, ``hash``,
+        ``config_key`` (optional). See Notes.
```

Comment:
so the reason it's nested is so that multiple archive files that are part of one "logical dataset" can be downloaded all at once, right?  I think that would also be achievable with a list of dicts, if you add a key "dataset_name" alongside "url", "hash", etc. That might be a nicer API for the single-file case:  users can pass  a single dict in that case. To me it seems a little cleaner, WDYT?

Reply:
Makes sense to me.

Should I change the format in `config.py`, or handle internally the MNE datasets to go from dict of dict -> list of dict if needed when calling `fetch_dataset`?

I think the format in `config.py` is nice cuz it consolidates LOC per dataset.